### DISCUSSION TITLE
can caches cause memory leak?

### DISCUSSION THREAD

**Discussion by mucahidyazar** [TanStack/query#3771] [Q&A] [answered]
If I use cacheTime as Infinity, this can cause a memory leak on the app?

**Comment by TkDodo** [comment, marked as answer]
the cache is just an object in javascript. We distribute it via react context through the `QueryClientProvider`. Infinite CacheTime means we'll never remove it from the cache. So the data will stick around until you unmount your `QueryClientProvider`.

Usually, in a single page application, you'll only have one `QueryClientProvider` at the top of your App. So that cache entry will stick around for the lifetime of your app.

I'm not really sure what a "memory leak" would be for you?

**Reply by mucahidyazar** [reply]
@TkDodo Thank you for your answer. I meant with "memory leak", the data will be added every time to somewhere and its size will get bigger each time with our caches and blood out. But I understood with your answer it is just a Javascript object and its size can't get more prominent as I thought. So thank you I got my answer.
